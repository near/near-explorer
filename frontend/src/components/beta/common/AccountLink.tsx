import * as React from "react";
import Link from "../../utils/Link";
import { shortenString } from "../../../libraries/formatting";
import { styled } from "../../../libraries/styles";

const AccountLinkWrapper = styled("a", {
  whiteSpace: "nowrap",
});

export interface Props {
  accountId: string;
}

const AccountLink: React.FC<Props> = React.memo(({ accountId }) => {
  return (
    <Link href={`/beta/accounts/${accountId}`} passHref>
      <AccountLinkWrapper>{shortenString(accountId)}</AccountLinkWrapper>
    </Link>
  );
});

export default AccountLink;
