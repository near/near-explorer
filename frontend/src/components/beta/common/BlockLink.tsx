import * as React from "react";
import Link from "../../utils/Link";
import { styled } from "../../../libraries/styles";

export interface Props {
  blockHash: string;
  blockHeight: number;
}

const LinkWrapper = styled("a", {
  whiteSpace: "nowrap",
  cursor: "pointer",
});

const BlockLink: React.FC<Props> = React.memo(({ blockHash, blockHeight }) => (
  <Link href="/blocks/[hash]" as={`/blocks/${blockHash}`}>
    <LinkWrapper>{`#${blockHeight}`}</LinkWrapper>
  </Link>
));

export default BlockLink;
