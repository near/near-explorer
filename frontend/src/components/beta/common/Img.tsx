import * as React from "react";
import Image from "next/image";
import { styled } from "../../../libraries/styles";
import { rgbDataURL } from "../../../libraries/rgbplaceholder";

const FallbackImage = styled("img", {
  width: "100%",
});

type Props = {
  src: string | null;
};
const Img: React.FC<Props> = React.memo(({ src }) => {
  const [mediaUrl, setMediaUrl] = React.useState(src ?? null);
  const handleMediaUrl = React.useCallback(
    (url) => () => setMediaUrl(url),
    [setMediaUrl]
  );

  if (!mediaUrl) {
    return <FallbackImage src="/static/images/failed_to_load.svg" />;
  }

  return (
    <Image
      src={mediaUrl}
      layout="fill"
      placeholder="blur"
      blurDataURL={rgbDataURL(220, 220, 220)}
      onError={handleMediaUrl("/static/images/failed_to_load.svg")}
      onLoadingComplete={handleMediaUrl(src)}
    />
  );
});

export default Img;
