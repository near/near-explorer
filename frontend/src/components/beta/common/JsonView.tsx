import dynamic from "next/dynamic";
import * as React from "react";
// https://github.com/mac-s-g/react-json-view/issues/296#issuecomment-803497117
const DynamicReactJson = dynamic(import("react-json-view"), { ssr: false });

type Props = {
  args: object;
};

const JsonView: React.FC<Props> = React.memo(({ args }) => (
  <DynamicReactJson
    src={args}
    name={null}
    iconStyle="triangle"
    displayObjectSize={false}
    displayDataTypes={false}
    style={{
      fontSize: "14px",
      padding: "10px 0",
    }}
  />
));

export default JsonView;
