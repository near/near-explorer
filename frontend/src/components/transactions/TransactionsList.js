import { useEffect } from "react";

import { Row, Col } from "react-bootstrap";

import { DataConsumer } from "../utils/DataProvider";
import EmptyRow from "../utils/EmptyRow";

import ActionRow from "./ActionRow";
import TransactionsHeader from "./TransactionsHeader";
import TransactionsPaginationFooter from "./TransactionsPaginationFooter";

const Transactions = ({ transactions }) => {
  return (
    <>
      {transactions.flatMap((transaction, transactionIndex) =>
        transaction.actions
          .reverse()
          .map((action, actionIndex) => (
            <ActionRow
              key={transaction.hash + actionIndex}
              transaction={transaction}
              action={action}
              cls={`${
                transactions.length - 1 === transactionIndex &&
                transaction.actions.length - 1 === actionIndex
                  ? "transaction-row-bottom"
                  : ""
              }`}
            />
          ))
      )}
      <EmptyRow />
      <TransactionsPaginationFooter />
      <EmptyRow rows="5" />
    </>
  );
};

export default Transactions;
