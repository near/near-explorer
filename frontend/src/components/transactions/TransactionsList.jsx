import ActionRow from "./ActionRow";

export default ({ transactions }) => {
  return (
    <>
      {transactions.flatMap((transaction, transactionIndex) =>
        transaction.actions
          .reverse()
          .map((action, actionIndex) => (
            <ActionRow
              key={transaction.hash + actionIndex}
              transaction={transaction}
              action={action}
              cls={`${
                transactions.length - 1 === transactionIndex &&
                transaction.actions.length - 1 === actionIndex
                  ? "transaction-row-bottom"
                  : ""
              }`}
            />
          ))
      )}
    </>
  );
};
