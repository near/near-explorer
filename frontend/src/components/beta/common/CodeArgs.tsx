import * as React from "react";
import { hexy } from "hexy";

import { styled } from "../../../libraries/styles";

import CodePreview from "../../utils/CodePreview";
import JsonView from "./JsonView";

const HexArgs = styled("div", {
  padding: "10px 0",

  "& > div": {
    background: "#f8f8f8",
    borderRadius: 4,
    color: "#3f4246",
    padding: 20,
    fontSize: 14,
    fontWeight: 500,

    "& textarea, pre": {
      background: "inherit",
      color: "inherit",
      fontFamily: "inherit",
      fontSize: "inherit",
      border: "none",
      padding: 0,
    },
  },
});

const CodeArgs: React.FC<{ args: string }> = React.memo(({ args }) => {
  const decodedArgs = Buffer.from(args, "base64");

  let prettyArgs: object | string;
  try {
    const parsedJSONArgs = JSON.parse(decodedArgs.toString());
    prettyArgs =
      typeof parsedJSONArgs === "boolean"
        ? JSON.stringify(parsedJSONArgs)
        : parsedJSONArgs;
  } catch {
    prettyArgs = hexy(decodedArgs, { format: "twos" });
  }
  return typeof prettyArgs === "object" ? (
    <JsonView args={prettyArgs} />
  ) : (
    <HexArgs>
      <CodePreview collapseHeight={200} maxHeight={600} value={prettyArgs} />
    </HexArgs>
  );
});

export default CodeArgs;
