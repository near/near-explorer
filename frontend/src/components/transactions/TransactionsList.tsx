import * as T from "../../libraries/explorer-wamp/transactions";

import ActionsList from "./ActionsList";
import { ViewMode } from "./ActionRow";

export interface Props {
  transactions: T.Transaction[];
  viewMode?: ViewMode;
}

export default ({ transactions, viewMode }: Props) => {
  return (
    <>
      {transactions.map(transaction => (
        <ActionsList
          key={transaction.hash}
          actions={transaction.actions.reverse()}
          transaction={transaction}
          viewMode={viewMode}
        />
      ))}
    </>
  );
};
