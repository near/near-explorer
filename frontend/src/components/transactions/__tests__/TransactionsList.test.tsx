import renderer from "react-test-renderer";

import TransactionsList from "../TransactionsList";

import { TRANSACTIONS } from "./common";

describe("<TransactionsList />", () => {
  it("renders", () => {
    expect(
      renderer.create(<TransactionsList transactions={TRANSACTIONS} />)
    ).toMatchSnapshot();
  });
});
