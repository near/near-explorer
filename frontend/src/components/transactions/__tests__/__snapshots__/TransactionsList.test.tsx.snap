// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TransactionsList /> renders 1`] = `null`;
